library/ssl_tls.c

```diff
@@ -5093,10 +5101,11 @@ static int ssl_compute_master( mbedtls_ssl_handshake_params *handshake,
     }
 #endif /* MBEDTLS_SSL_EXTENDED_MS_ENABLED */
 
-#if defined(MBEDTLS_USE_PSA_CRYPTO) &&                 \
-    ( defined(MBEDTLS_KEY_EXCHANGE_PSK_ENABLED) ||     \
-      defined(MBEDTLS_KEY_EXCHANGE_RSA_PSK_ENABLED) || \
-      defined(MBEDTLS_KEY_EXCHANGE_ECDHE_PSK_ENABLED) )
+#if defined(MBEDTLS_USE_PSA_CRYPTO) &&                   \
+    ( defined(MBEDTLS_KEY_EXCHANGE_PSK_ENABLED) ||       \
+      defined(MBEDTLS_KEY_EXCHANGE_RSA_PSK_ENABLED) ||   \
+      defined(MBEDTLS_KEY_EXCHANGE_ECDHE_PSK_ENABLED) || \
+      defined(MBEDTLS_KEY_EXCHANGE_DHE_PSK_ENABLED) )
     if( ( handshake->ciphersuite_info->key_exchange == MBEDTLS_KEY_EXCHANGE_PSK ||
```

Comment:
Pre-existing: the first parenthesis could be shortened to `mbedtls_ssl_ciphersuite_uses_psk( handshake->ciphersuite_info )` I think.